addon/index.js

```diff
@@ -538,20 +558,13 @@ export function changeset(obj, validateFn = defaultValidatorFn, validationMap =
         this._deleteKey(ERRORS, key);
 
         if (!isEqual(oldValue, value)) {
-          this._recursivelySet(key, value, changes);
+          deepSet(changes, key, value);
         } else if (key in changes) {
           delete changes[key];
         }
-
         this.notifyPropertyChange(CHANGES);
         this.notifyPropertyChange(root);
```

Comment:
But out of an `N` deep object graph, root will always be level `1`. Shouldn't it be notifying level `N-1`?